web/packages/teleport/src/Notifications/Notifications.tsx

```diff
@@ -276,6 +363,111 @@ function EmptyState() {
   );
 }
 
+/** accessListStoreNotificationsToNotifications converts a list of access list notifications from the notifications store into the primary
+ * Notification type used by the notifications list.
+ */
+function accessListStoreNotificationsToNotifications(
+  accessListNotifs: AccessListNotification[],
+  notificationStates: LocalNotificationStates
+): NotificationType[] {
+  const today = new Date();
+
+  /** dueNotifications are the notifications for access lists which are due for review in the future. */
+  const dueNotifications = accessListNotifs
+    .filter(notif => isAfter(notif.date, today))
+    // Sort by earliest dates.
+    .sort((a, b) => {
+      return a.date.getTime() - b.date.getTime();
+    });
+
+  /** overdueNotifications are the notifications for access lists which are overdue for review. */
+  const overdueNotifications = accessListNotifs
+    .filter(notif => isBefore(notif.date, today))
+    .sort((a, b) => {
+      return a.date.getTime() - b.date.getTime();
+    });
+
+  const processedDueNotifications: NotificationType[] = [];
+  const processedOverdueNotifications: NotificationType[] = [];
+
+  // If there are 2 or less access list notifications due for review, then we will return a notification per access list.
+  // If there are more than 2, then we return one "grouped" notification for all of them. This is to prevent clutter in the notifications list
+  // in case there are many access lists due for review.
+  if (dueNotifications.length <= 2) {
+    // Process and add them to the final processed array.
+    dueNotifications.forEach(notif => {
+      const numDays = formatDistanceToNowStrict(notif.date);
+      const titleText = `Access list '${notif.item.resourceName}' needs your review within ${numDays}.`;
+
+      processedDueNotifications.push({
+        localNotification: true,
+        title: titleText,
+        id: notif.id,
+        subKind: StoreNotificationKind.AccessList,
+        clicked: notif.clicked,
+        createdDate: today,
+        labels: [{ name: 'redirect-route', value: notif.item.route }],
+      });
+    });
+  } else {
+    const mostUrgentNotif = dueNotifications[0];
+    const numDays = formatDistanceToNowStrict(mostUrgentNotif.date);
+    const titleText = `${dueNotifications.length} of your access lists require review, the most urgent of which is due in ${numDays}.`;
+
+    // The ID for this combined notification is the ID's of all the notifications within it joined together.
+    const id = dueNotifications.map(notif => notif.id).join(',');
```

Comment:
this can potentially be a really long string 🤔 like think of 400 x UUID of length `edb2d3a8-aaaa-aaaa-9aaaa-f3f14681aaaa`

can we use like `crypto.randomUUID()`?

Reply:
We can't use a random UUID because this will be run on every page refresh and so the UUID for the gnotification will be different each time (and thus break the clicked/seen functionality). What I'll do as a middle ground is some type of rudimentary hashing where the ID of the group notification will be `<first notification in the list id>-<last notification in the list id>-<length of list>`

Reply:
was this addressed?